### ISSUE TITLE
BUG:  FaceLogin 1.8.1.3 CRASH when no users

### ISSUE THREAD

**Issue by JackMostow** [RoboTutorLLC/RTFace_Login#22]
On Wed, Aug 31, 2022 at 11:52 PM Rohan Kulkarni <[rohan.kulkz@gmail.com](mailto:rohan.kulkz@gmail.com)> wrote:
Hi Professor,

FaceLogin seems to be working well. I am unfortunately still having trouble building a release apk. 
It seems that there is a problem with the signing credentials.

Rohan - 
1. Can you build a Debug apk? --> NO unless tethered.
2. Any reason not to deploy it? --> See 1.
3. I don't know how signing credentials work.  Would a token help? --> No, but keystore.properties should.

Next issue:  need way to reset database to no-users state.
Solution:  put in a line to delete all users just to debug.  Once fixed, comment it out.

**Comment by JackMostow** [comment]
BTW, once this works, also please:

1. Use the simple nested query to order the animals by popularity (# users with that animal, if any) to keep everyone from picking the same animal, then by RAND() to break ties randomly so that even different tablets with zero users don't list the same animal first.

2. Log state changes in FaceLogin to facilitate future debugging (if necessary) and to document login behavior (if informative).

BUT FIRST GET A WORKING APK!

**Comment by JackMostow** [comment]
@madeleinvillegas - Thanks for posting 1.8.1.3 apk [here](https://drive.google.com/file/d/1h-sJ3VzgHLes6kOqBLtqRXVsnW_YtkJQ/view?usp=sharing) since it's 27MB, supposedly too big to post on GitHub.
I enrolled and logged in, rejecting selfie and animal photo before accepting them.
It crashed (without leaving a crash log) only when I tried to go on at the end and FaceLogin tries to launch RoboTutor, which wasn't installed on my tablet.  Ideally it should display an informative error message at that point, but ideally it shouldn't get to that point because we expect RoboTutor to be already installed on the tablet.
